### DISCUSSION TITLE
How does Dokku handle "dynos"?

### DISCUSSION THREAD

**Discussion by JeremyEnglert** [dokku/dokku#5216] [General]
Hi,

We're thinking of moving from Heroku to Dokku because our app is pretty memory heavy and we can't afford $250/mo dynos.

How does Dokku handle the concept of "dynos"?

For example, right now, we have 1 web dyno and 2 worker dynos. When we create those, we're able to decide the size of those dynos.

When you add a worker on Dokku, do you pick a "size" or do all workers share whatever resources are available on the server/droplet?

**Comment by josegonzalez** [comment]
In Dokku, a Dyno is effectively a container. Processes can be scaled up or down, and each instance of a process runs in it's own container. All containers have - by default - access to all available system resources (so the system is shared) using the docker-local scheduler (there are others, such as kubernetes and nomad, as noted in the docs). You can limit/reserve system resources for processes using the built-in `resources` plugin (as mentioned in our docs), and each container process gets those resource limits/reservations attached to it (whether you have them available is another story).

Hope that makes sense.
